src/cmd/linuxkit/pkglib/build.go

```diff
@@ -191,3 +201,62 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 
 	return nil
 }
+
+type buildCtx struct {
+	sources []string
+}
+
+// Copy iterates over the sources, tars up the content after rewriting the paths.
+// It assumes that sources is sane, ie is well formed and the first part is an absolute path
+// and that it exists. NewFromCLI() ensures that.
+func (c *buildCtx) Copy(w io.WriteCloser) {
+	tw := tar.NewWriter(w)
+	defer func() {
+		tw.Close()
+		w.Close()
+	}()
+
+	for _, source := range c.sources {
+		tmp := strings.Split(source, ":")
+		src := tmp[0]
+		dst := tmp[1]
+		log.Debugf("Adding to build context: %s -> %s", src, dst)
+
+		f := func(p string, i os.FileInfo, err error) error {
+			if err != nil {
+				log.Warnf("ctx: Walk error on %s: %v", p, err)
+				return nil
+			}
+
+			if i.IsDir() {
+				return nil
```

Comment:
I guess the untar on the other end automatically creates missing dirs, but aren't you throwing away ownership and perms etc? Is there any harm in outputting the directories too?

I'm not entirely sure that the remaining code handles symlinks and the like ok, it seems to either collapse them to files or output a symlink header followed by some actual data, not sure what the result is.

Reply:
Let me try to refactor this a little bit. I got an error when hitting directories, but I think I now know a better way to handle this